File: week 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // Get text input
    string text = get_string("Text: ");

    int letters = 0;
    int words = 0;
    int sentences = 0;

    // Count letters, words, and sentences
    for (int i = 0; text[i] != '\0'; i++)
    {
        char c = text[i];

        // Count letters
        if (isalpha(c))
        {
            letters++;
        }

        // Count words
        if (isspace(c) && text[i+1] != '\0' && !isspace(text[i+1]))
        {
            words++;
        }

        // Count sentences (. ! ?)
        if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    // Add 1 to words to count the last word
    words++;

    // Compute averages per 100 words
    float lettersAvg = ((float) letters / words) * 100;
    float sentencesAvg = ((float) sentences / words) * 100;

    // Compute Coleman-Liau index
    int grade = round(0.0588 * lettersAvg - 0.296 * sentencesAvg - 15.8);

    // Print grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: week 6/sentimental-readability/readability.py
from cs50 import get_string

# Get text input
text = get_string("Text: ")

letters = 0
words = 0
sentences = 0

# Count letters, words, and sentences
for i, c in enumerate(text):
    # Count letters
    if c.isalpha():
        letters += 1

    # Count words
    if c.isspace() and i + 1 < len(text) and not text[i + 1].isspace():
        words += 1

    # Count sentences (. ! ?)
    if c in ".!?":
        sentences += 1

words += 1  # Last word

# Compute averages per 100 words
letters_avg = (letters / words) * 100
sentences_avg = (sentences / words) * 100

# Compute Coleman-Liau index
grade = round(0.0588 * letters_avg - 0.296 * sentences_avg - 15.8)

# Print grade
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")
